### DISCUSSION TITLE
How density dependence is implemented in RangeShitR

### DISCUSSION THREAD

**Discussion by anyperez** [RangeShifter/RangeshiftR-tutorials#7] [Q&A] [answered]
Hi everyone,

I'm doing a stage-structured only-female model for a single closed population on the RangeShiftR package. As there is no dispersal, I'm using only one cell on the landscape module with a resolution of 100 (the code actually shows an error when I try to use a greater resolution, not sure why). The initial pop size would be 84,000 females. I want to introduce a carrying capacity (or 1/b parameter for the stage-structured model) equal to twice the initial pop size i.e. K_or_DensDep = 2*84000, to be used for the density dependence in survival of stages 1 and 2, and the fecundity of stage 3.
Given this parameterization, my script is the following:

sim <- Simulation(Simulation = 1, Years = 50, Replicates = 1)
land <- ArtificialLandscape(propSuit = 1, K_or_DensDep = 2*84000, Resolution = 100,  dimX = 1, dimY = 1)
stg <- StageStructure(Stages = 4, TransMatrix = matrix(c(0,.67,0,0,0,.744,.186,0,.49,0,.939,.041,.05,0,0,.71),nrow = 4), MaxAge = 50, MinAge = c(0,0,6,30), SurvDensDep = TRUE, FecDensDep = TRUE, SurvStageWtsMatrix = matrix(c(1,1,1,1,1,1,1,1,0,0,0,0,0,0,0,0),nrow = 4), FecStageWtsMatrix = matrix(c(0,0,0,0,0,0,0,0,1,1,1,1,0,0,0,0),nrow = 4))
demo <- Demography(StageStruct = stg)
init <- Initialise(InitDens = 2, IndsHaCell = 84000, PropStages = c(0,.34,.58,.08))
s <- RSsim(simul = sim, land = land, demog = demo, init = init)
validateRSparams(s)
RunRS(s, dirpath)

I would expect that the exponential decay that describes the density dependence in survival and fecundity would ONLY act on the population when it is approaching the carrying capacity in order to regulate the population size when it gets too large. However, when I run the simulation and plot the abundance over time, the population size decreases right from the start without ever increasing nor getting close to K. I get similar outcomes when I try to increase K, but even if the population rises during the first years of the simulation, it then decreases as if the density dependence would be acting at all times independently of the population reaching K. So, **I'd like to know how the density dependence in the demographic parameters is implemented in the package (besides its mathematical form) or if I'm making something wrong/missing something important to get the desired model.**

Many thanks for your help

**Comment by damariszurell** [comment, marked as answer]
Hi Any,

You can look at the form of the density dependence using the function plotProbs(). For example, look at the description of the demography module in Tutorial 2 https://rangeshifter.github.io/RangeshiftR-tutorials/tutorial_2.html#21_Demographic_and_dispersal_parameters.

**Reply by anyperez** [reply]
Thank you Damaris. After plotting the probabilities with plotProbs(demo@StageStruct), I notice that the density dependence is acting on the population for any population size, i.e. at all time steps. My question is, if it is possible to apply the density dependence in the demographic parameters ONLY when the population size is reaching the carrying capacity, so that the population doesn't decrease so fast.

**Reply by JetteReeg** [reply]
Hi Any, you also need to have in mind, that - as you are working with a stage structured model - your density depedence is defined by the parameter 1/b and not K. Have you explored the equilibrium population size as shown in tutorial 3 (https://rangeshifter.github.io/RangeshiftR-tutorials/tutorial_3.html#222_Set_landscape_and_explore_equilibrium_population_size)?

**Reply by damariszurell** [reply]
Hi, RangeShifter has not implemented a ceiling effect for carrying capacity. You can manipulate the decay of the density dependence to some extent using DevDensCoeff and SurvDensCoeff.

**Reply by anyperez** [reply]
> Hi Any, you also need to have in mind, that - as you are working with a stage structured model - your density depedence is defined by the parameter 1/b and not K. Have you explored the equilibrium population size as shown in tutorial 3 (https://rangeshifter.github.io/RangeshiftR-tutorials/tutorial_3.html#222_Set_landscape_and_explore_equilibrium_population_size)?

Hi there. Thanks for the suggestion. Yes, I've used the function getLocalisedEquilPop() but yet the population keeps decreasing at a high rate even using a more appropriate 1/b value, and it isn't approching the maximum population abundance that I would expect.


> Hi, RangeShifter has not implemented a ceiling effect for carrying capacity. You can manipulate the decay of the density dependence to some extent using DevDensCoeff and SurvDensCoeff.

Hi Damaris, thank you for your answer. I'll do some simulations with different coefficients to see how it goes.

**Comment by anyperez** [comment]
Hi everyone,

Sorry to insist on the same topic, but the results I get still don't make any sense, so I'd like to know if I'm doing something wrong and I hope you can help me identify what the problem is. Based on the situation that I want to model (described above, together with the code used), I initially ran a null model (without density dependence) for which the mean population growth was 7%. However, once I incorporated the density dependence (as shown in the code above), the population declined from the first timestep as the following graph shows:
![Sim29_SurvAndFecDensDep_K2_C1_onlySurvpupandjuvFecadult](https://user-images.githubusercontent.com/89428629/134043243-623550d9-84ce-4078-91a8-89ec41a682ae.png)

This seems pretty odd to me. The whole point about density dependence is that the pop goes to an equilibrium at the carrying capacity.  In the null model which grows without the density dependence, it should continue to grow with density dependence until it approaches k, when growth should start to slow, and then it goes to an equilibrium with lamba=1 when k is reached.  The pop shouldn’t decrease for the null model vital rates because of density dependence. So, **what might be wrong about the code?** Besides, in a stage-structured model where K_or_DensDep is the 1/b parameter that describes the strength of dens dep, **where can the carrying capacity of the model be specified?**

I really appreciate your help on this matter.

**Reply by damariszurell** [reply]
Hi there,
Again, I need to point you towards re-reading the RangeShifter Manual and tutorials. In the case of stage-structured models, the parameter 1/b (K_or_DensDep in RangeShiftR) does not refer to carrying capacity but to the form of density dependence (see Neubert, M., Caswell, H. Density-dependent vital rates and their population dynamic consequences. J Math Biol 41, 103–121 (2000). https://doi.org/10.1007/s002850070001). The form of this density dependence you can easily visualise using plotProbs(). For understanding how the selected 1/b in your model affects demography and what equilibrium density sizes emerge from this, you can use the function getLocalisedEquilPop(). If you explore this a little, you will see that for obtaining equilibrium densities around 84000 individuals as you seem to want, then you need a 1/b of c. 250000 individuals/ha.
